Add household pages

CJWaszczuk opened the pull request.

CJWaszczuk committed:
Add rent arrears page

Still needs post-visit action checkboxes.
Also needs to have dynamic rent arrears from API.

CJWaszczuk committed:
Add other property page

Needs post-visit action checkbox.

erbridge commented on steps/household/rent.tsx:
```diff
@@ -0,0 +1,343 @@
+            ) as React.ReactNode,
+            radios: [
+              {
+                label: "Yes and payments received",
```
```suggestion
                label: "Yes, and payments received",
```

erbridge commented on steps/household/rent.tsx:
```diff
@@ -0,0 +1,343 @@
+                value: "no wants to apply"
+              },
+              {
+                label: "No, has not applied",
```
```suggestion
                label: "No, and does not want to apply",
```

erbridge commented on steps/household/review-household.tsx:
```diff
@@ -0,0 +1,142 @@
+                  type={ListTypes.Bullet}
+                  items={[
+                    <>
+                      <Link href="/">Downsizing</Link>: A payment made for
```
These `Link`s need to be the `lbh-frontend-react` version, not the `next/link` version.

erbridge commented on steps/household/review-household.tsx:
```diff
@@ -0,0 +1,142 @@
+                      <Link href="">Seaside and country</Link> if aged over 60
+                    </>
+                  ]}
+                ></List>
```
```suggestion
                />
```

Prefer self-closing.

erbridge commented on steps/household/review-household.tsx:
```diff
@@ -0,0 +1,142 @@
+                      <Link href="/">Overcrowding</Link>
+                    </>,
+                    <>
+                      <Link href="">Seaside and country</Link> if aged over 60
```
Do we know where these should go? Can we put them in? Also, make sure they open in a new tab.

LBHELewis replied:
Do we definitely want to open in a new tab? This goes against the latest accessibility advice as far as I know?

CJWaszczuk replied:
Have been looking at https://www.w3.org/TR/WCAG20-TECHS/G200.html. For the case of online forms (https://www.w3.org/TR/WCAG20-TECHS/G200.html#G200-ex1), it's suggested that they be redirected away for excessive context-sensitive.

erbridge replied:
We also want to avoid encouraging users to navigate away from the process at all, so any external links should be in new tabs for that reason.

erbridge commented on steps/household/review-household.tsx:
```diff
@@ -0,0 +1,142 @@
+          // eslint-disable-next-line react/display-name
+          Component: (props): JSX.Element => {
+            return (
+              <Details summary="Housing move schemes: downsizing, overcrowding or aged over 60">
```
```suggestion
              <Details summary="House moving schemes: downsizing, overcrowding or aged over 60">
```

erbridge commented on steps/household/review-household.tsx:
```diff
@@ -0,0 +1,142 @@
+          props: {
+            name: "housing-move-schemes-notes",
+            label: {
+              value: "Add note about any interest in housing move schemes"
```
```suggestion
              value: "Add note about any interest in house moving schemes"
```

erbridge commented on steps/household/review-household.tsx:
```diff
@@ -0,0 +1,142 @@
+          >({
+            storeName: "household",
+            key: processRef,
+            property: ["housingMoveSchemes", "notes"]
```
```suggestion
            property: ["houseMovingSchemes", "notes"]
```

erbridge commented on steps/household/review-household.tsx:
```diff
@@ -0,0 +1,142 @@
+            headings: ["Full name", "Relationship to tenant", "Date of birth"],
+            rows: [
+              ["John Smith", "Father", "10/04/1967"],
+              ["Steve Davis", "Uncle", "12/03/1978"]
```
The data for this exists in the `ExternalDatabase` now. Let's use it.

erbridge commented:
```
/home/circleci/project/steps/household/rent.tsx
  16:10  warning  'Details' is defined but never used     @typescript-eslint/no-unused-vars
  17:10  warning  'ImageInput' is defined but never used  @typescript-eslint/no-unused-vars
  20:10  warning  'TextArea' is defined but never used    @typescript-eslint/no-unused-vars
```

erbridge commented on storage/Storage.ts:
```diff
@@ -56,6 +56,7 @@ export default class Storage {
           if (version === 0) {
             upgrade.createStore("lastModified");
             upgrade.createStore("property");
+            upgrade.createStore("household");
```
I think we need to start bumping the version now, since we're deploying somewhere non-developers are interacting with it.

erbridge committed:
Drop the `null` type from optional props

It's confusing, makes working out type casting difficult, and we don't
handle `null` any differently from `undefined`.

erbridge committed:
Allow `TextAreaDetails` to specify content before the textarea

erbridge committed:
Create a `Table` component

erbridge committed:
Add the review household members step

This bumps the schema version to version 2, as we now have
non-developers interacting with the service, and need to support
graceful upgrades.

erbridge committed:
Update the sections page to show the household status

erbridge commented:
@LBHELewis Mind taking another look at this, please?
